portals/application/modules/provision/controllers/IndexController.php

```diff
@@ -33,70 +24,56 @@ public function indexAction()
+        }
+
+        if (!$isHttps) {
+            return $this->_error(403);
+        }
+
+        $terminal = $this->_getTerminalByUrl($terminalUrl);
+        if (!$terminal) {
+            return $this->_error(404, 'Terminal not found');
+        }
 
-       $terminalUrl = $this->getRequest()->getParam('requested_url');
-
-       $path = $this->_getFilePath();
-
-       $terminalMapper = new \IvozProvider\Mapper\Sql\TerminalModels();
-       $terminalModel = $this->_searchGenericPattern($terminalMapper, $terminalUrl);
-
-       if ( $terminalModel == null ) {
-
-           if (!isset($_SERVER['HTTPS']) || $_SERVER['HTTPS'] != "on") {
-               $this->_error('Trying to access using incorrect protocol', 'Forbidden', 403);
-               return;
-           }
-
-           $data = $this->_searchSpecificPattern($terminalMapper, $terminalUrl);
-           if($data) {
-               $terminalModel = $data['terminalModel'];
-               $urlVariables = $data['urlVariables'];
-               $userMapper = new \IvozProvider\Mapper\Sql\Users();
-
-               foreach( $this->_allowedVariables as $key => $variable){
-                   $mapper = new $variable['mapperName']();
-                   $model = $mapper->findOneByField($variable['field'], $urlVariables[$key]);
-
-                   if ($model) {
-                       if($model instanceof IvozProvider\Model\Terminals){
-                           $userModel = $userMapper->findOneByField('terminalId', $model->getId() );
-                           $this->view->user = $userModel;
-
-                           $companyModel = $model->getCompany();
-                           $this->view->company = $companyModel;
-
-                           $brandModel = $companyModel->getBrand();
-                           $this->view->brand = $brandModel;
-
-                           $this->view->terminal = $model;
-                       }
-
-                       $viewKey = $variable['viewName'];
-                       $this->view->{$viewKey} = $model;
-                   }
-                   else{
-                       $this->logger->log($variable['mapperName'] . $variable['field'] . ' = '. $urlVariables[$key] .' does not exist', Zend_Log::WARN);
-                   }
-               }
-
-               $this->_renderPage('specific', $path, $terminalModel);
-           }
-           else{
-               $this->_error('TerminalModel not found', 'Not found', 200);
-           }
-       }
-       else{
-           $this->_renderPage('generic', $path, $terminalModel);
-       }
+        $terminalModel = $terminal->getTerminalModel();
+        if (!$terminalModel) {
+            return $this->_error(404, 'TerminalModel not found');
+        }
+
+        $userMapper = new \IvozProvider\Mapper\Sql\Users();
+        $userModel = $userMapper->findOneByField('terminalId', $terminal->getId() );
+        $this->view->user = $userModel;
```

Comment:
Won't `$terminal->getUser()` work here? It's declared in the _smart_ terminal's model.